src/pm/hydra/lib/tools/bootstrap/external/slurm_env.c

```diff
@@ -14,9 +14,9 @@ HYD_status HYDT_bscd_slurm_query_env_inherit(const char *env_name, int *ret)
     /* check if envvar starts with SBATCH_, SLURMD_, or SLURM_ */
     if (strncmp(env_name, "SBATCH_", 7) == 0 || strncmp(env_name, "SLURMD_", 7) == 0 ||
         strncmp(env_name, "SLURM_", 6) == 0) {
-        *ret = 1;
-    } else {
         *ret = 0;
+    } else {
+        *ret = 1;
```

Comment:
The choice of var name `ret` was the source of confusion. It gives no clue on its logic. How about rename it `should_inherit` (for every `query_env_inherit` instances)?

Reply:
Good idea. Will add a commit.

Reply:
Added. I also redid a loop I found confusing 😅.